app/models/user.rb

```diff
@@ -5,4 +5,13 @@ class User < ActiveRecord::Base
          :recoverable, :rememberable, :trackable, :validatable
 
   has_many :comments
+  has_many :requests
+
+  def requested_book?(book)
+    self.active_book_requests.exists?(book_id: book.id)
+  end
+
+  def active_book_requests
```

Comment:
I think `pending_book_requests` is more accurate, since this could be confusing.
Also, could this be private?